### PR TITLE
sbendus - create test

### PR DESCRIPTION
**Pull request by SofiiaPro**

### CODE REVIEW

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 1]
```diff
@@ -1,20 +1,39 @@
...
                 .openHomePage();
     }
 
+    @Test
+    public void testSelectProductByCatalog() {
+        String pageCategoryName = "Ноутбуки";
+        CategoryPage catalogCategoryItem = rozetkaHomePage
+                .getLeftSidebar()
+                .openCategory(pageCategoryName);
+        Assert.assertTrue(catalogCategoryItem.getPageTitle().contains("Комп'ютери"),
```
Comment (VladimirPetrat) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java:27]: Add string to local variable

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 2]
```diff
@@ -1,20 +1,39 @@
...
+                "The page title should match the condition");
+        ProductPage chosenProduct = catalogCategoryItem
+                .openProductsListPage(pageCategoryName)
+                .openProductByName("Ноутбук Apple MacBook Air 13");
+        Assert.assertTrue(chosenProduct.getProductTitle().contains("Ноутбук Apple MacBook"),
+                "The product title should match the condition");
+        chosenProduct.openCategoryPageByName("ноутбуки");
+        Assert.assertTrue(catalogCategoryItem.getPageTitle().contains("Комп'ютери"),
```
Comment (VladimirPetrat) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: Problem with mapping

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 3]
```diff
@@ -1,20 +1,39 @@
...
+                .getLeftSidebar()
+                .openCategory(pageCategoryName);
+        Assert.assertTrue(catalogCategoryItem.getPageTitle().contains("Комп'ютери"),
+                "The page title should match the condition");
+        ProductPage chosenProduct = catalogCategoryItem
+                .openProductsListPage(pageCategoryName)
+                .openProductByName("Ноутбук Apple MacBook Air 13");
+        Assert.assertTrue(chosenProduct.getProductTitle().contains("Ноутбук Apple MacBook"),
+                "The product title should match the condition");
+        chosenProduct.openCategoryPageByName("ноутбуки");
```
Comment (VladimirPetrat) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: CategoryPage catalogCategoryItemAgain = chosenProduct.openCategoryPageByName("ноутбуки");

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/WishListPage.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+import static com.codeborne.selenide.Selenide.$x;
+
+public class WishListPage {
+
+    public String getProductTitleFromWishProductList(String message) {
+        return $x(String.format("//span[@class='goods-tile__title'] and contains(text()), '%s'", message))
+                .text();
+    }
+}
+
```
Comment (SofiiaPro) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/WishListPage.java]: //STILL IN PROGRESS.
Applies to the SECOND test case.

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/LeftSidebar.java [hunk 5]
```diff
@@ -10,4 +10,11 @@ public CategoryPage openCategory(String categoryName) {
         $x("//button[@class='search-form__microphone']").click();
         return new CategoryPage();
     }
+
+    public CategoryPage openSalesCategory() {
+        $x("//a[@class = 'menu-categories__link menu-categories__item_type_bordered']").click();
+        // click on microphone element to hide dropdown
+        $x("//button[@class='search-form__microphone']").click();
+        return new CategoryPage();
+    }
```
Comment (SofiiaPro) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/LeftSidebar.java]: //STILL IN PROGRESS.
Applies to the SECOND test case.

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java [hunk 6]
```diff
@@ -30,4 +32,7 @@ public String getPageTitle() {
         return $x("//h1[@class='portal__heading']").text();
     }
 
+    public String getPageUrlTitle() {
+        return Selenide.title();
+    }
```
Comment (SofiiaPro) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java]: misunderstood with page title (header in the page) and page URL title.

divide these into:
page title - string title on the page
URL title - title in URL

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 7]
```diff
@@ -50,4 +50,32 @@ public int getQuestionListSize() {
...
+    }
+
+    public Integer getProductWithSale() {
+        String productWithSaleText = $x("//p[@class = 'product-prices__big product-prices__big_color_red']")
+                .text();
+        Integer productWithSale = Integer.parseInt(productWithSaleText
+                .substring(0, productWithSaleText.length() - 1)
+                .replaceAll(" ", ""));
+        return productWithSale;
+    }
```
Comment (SofiiaPro) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: //STILL IN PROGRESS.
Applies to the SECOND test case.

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java [hunk 8]
```diff
@@ -30,4 +32,7 @@ public String getPageTitle() {
         return $x("//h1[@class='portal__heading']").text();
     }
 
+    public String getPageUrlTitle() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/CategoryPage.java]: that's not url

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 9]
```diff
@@ -50,4 +50,32 @@ public int getQuestionListSize() {
...
+        $x("//app-goods-wishlist[@class='product__favorites']//button[@class='wish-button js-wish-button']")
+                .doubleClick();
+        return this;
+    }
+
+    public WishListPage openWishListPage() {
+        return new WishListPage();
+    }
+
+    public Integer getProductOldPrice() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: old means pre-discount?

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java [hunk 10]
```diff
@@ -50,4 +50,32 @@ public int getQuestionListSize() {
...
+        String productOldPriceText = $x("//p[@class = 'product-prices__small']")
+                .text();
+        Integer productOldPrice = Integer.parseInt(productOldPriceText
+                .substring(0, productOldPriceText.length() - 1)
+                .replaceAll(" ", ""));
+        return productOldPrice;
+    }
+
+    public Integer getProductWithSale() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/ProductPage.java]: rename

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/WishListPage.java [hunk 11]
```diff
@@ -0,0 +1,12 @@
+package com.softserveinc.ita.rozetka.page_objects;
+
+import static com.codeborne.selenide.Selenide.$x;
+
+public class WishListPage {
+
+    public String getProductTitleFromWishProductList(String message) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/WishListPage.java]: rename all

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 12]
```diff
@@ -1,20 +1,41 @@
...
 
-    //This added for future tests
     @BeforeMethod
     public void openHomepage() {
         rozetkaHomePage = new HomePage()
                 .openHomePage();
     }
 
+    @Test
+    public void testSelectProductByCatalog() {
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java:65]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 13]
```diff
@@ -1,20 +1,41 @@
...
+        String pageCategoryName = "Ноутбуки";
+        String pageTitle = "Комп'ютери";
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: declare and use where needed, not before

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 14]
```diff
@@ -1,20 +1,41 @@
...
+        CategoryPage chosenCategoryByCataloge = rozetkaHomePage
+                .getLeftSidebar()
+                .openCategory(pageCategoryName);
+        Assert.assertTrue(chosenCategoryByCataloge.getPageTitle().contains(pageTitle),
+                "The page title should match the condition");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: Incorrect page title

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Review by toleksyn — APPROVED** [review]

File: src/main/java/com/softserveinc/ita/rozetka/page_objects/LeftSidebar.java [hunk 15]
```diff
@@ -11,4 +11,11 @@ public CategoryPage openCategory(String categoryName) {
         $x("//button[@class='search-form__microphone']").click();
         return new CategoryPage();
     }
+
+    public CategoryPage openSalesCategory() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/ita/rozetka/page_objects/LeftSidebar.java]: too specific

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 16]
```diff
@@ -41,5 +41,43 @@ public void testSelectProductBySearch() {
         assertTrue(productTitle.contains(searchRequest),
                 "Product title should contain search request");
     }
+
+    @Test
+    public void testSelectProductByCatalog() {
+        var pageCategoryName = "Ноутбуки";
+        var chosenCategoryByCatalog = rozetkaHomePage
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 17]
```diff
@@ -41,5 +41,43 @@ public void testSelectProductBySearch() {
...
+                "Incorrect page title");
+        var chosenProduct = chosenCategoryByCatalog
+                .openProductsListPage(pageCategoryName)
+                .openProductByName("Ноутбук Apple MacBook Air 13");
+        var isProductTitleCorrect = chosenProduct
+                .getProductTitle()
+                .contains("Ноутбук Apple MacBook");
+        assertTrue(isProductTitleCorrect, "Incorrect product title");
+        var chosenCategoryByLink = chosenProduct
+                .openCategoryPageByName(pageCategoryName.toLowerCase());
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java:81]: rename, fix

File: src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java [hunk 18]
```diff
@@ -41,5 +41,43 @@ public void testSelectProductBySearch() {
...
+        var catalogCategoryItem = rozetkaHomePage
+                .getLeftSidebar()
+                .openSalesCategory();
+        var isPageTitleCorrect = catalogCategoryItem
+                .getPageTitle()
+                .contains("Знижки");
+        assertTrue(isPageTitleCorrect, "Incorrect page title");
+        var productPage = catalogCategoryItem.openProductByNumber(1);
+        var productWithPreDiscountPrice = productPage.getPreDiscountPrice();
+        var productWithDiscountPrice = productPage.getDiscountPrice();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/ita/rozetka_test/RozetkaProductNavigationTest.java]: rename both

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Pull request merged**